common/persistence/sql/sqlExecutionManager.go

```diff
@@ -183,102 +184,92 @@ func (m *sqlExecutionManager) GetWorkflowExecution(request *p.GetWorkflowExecuti
 		DomainID:                     execution.DomainID.String(),
 		WorkflowID:                   execution.WorkflowID,
 		RunID:                        execution.RunID.String(),
-		TaskList:                     execution.TaskList,
-		WorkflowTypeName:             execution.WorkflowTypeName,
-		WorkflowTimeout:              int32(execution.WorkflowTimeoutSeconds),
-		DecisionTimeoutValue:         int32(execution.DecisionTaskTimeoutMinutes),
-		State:                        int(execution.State),
-		CloseStatus:                  int(execution.CloseStatus),
-		LastFirstEventID:             execution.LastFirstEventID,
 		NextEventID:                  execution.NextEventID,
-		LastProcessedEvent:           execution.LastProcessedEvent,
-		StartTimestamp:               execution.StartTime,
-		LastUpdatedTimestamp:         execution.LastUpdatedTime,
-		CreateRequestID:              execution.CreateRequestID,
-		DecisionVersion:              execution.DecisionVersion,
-		DecisionScheduleID:           execution.DecisionScheduleID,
-		DecisionStartedID:            execution.DecisionStartedID,
-		DecisionRequestID:            execution.DecisionRequestID,
-		DecisionTimeout:              int32(execution.DecisionTimeout),
-		DecisionAttempt:              execution.DecisionAttempt,
-		DecisionTimestamp:            execution.DecisionTimestamp,
-		StickyTaskList:               execution.StickyTaskList,
-		StickyScheduleToStartTimeout: int32(execution.StickyScheduleToStartTimeout),
-		ClientLibraryVersion:         execution.ClientLibraryVersion,
-		ClientFeatureVersion:         execution.ClientFeatureVersion,
-		ClientImpl:                   execution.ClientImpl,
-		SignalCount:                  int32(execution.SignalCount),
-		HistorySize:                  execution.HistorySize,
-		CronSchedule:                 execution.CronSchedule,
+		TaskList:                     info.GetTaskList(),
+		WorkflowTypeName:             info.GetWorkflowTypeName(),
+		WorkflowTimeout:              info.GetWorkflowTimeoutSeconds(),
+		DecisionTimeoutValue:         info.GetDecisionTaskTimeoutMinutes(),
+		State:                        int(info.GetState()),
+		CloseStatus:                  int(info.GetCloseStatus()),
+		LastFirstEventID:             info.GetLastFirstEventID(),
+		LastProcessedEvent:           info.GetLastProcessedEvent(),
+		StartTimestamp:               time.Unix(0, info.GetStartTimeNanos()),
+		LastUpdatedTimestamp:         time.Unix(0, info.GetLastUpdatedTimeNanos()),
+		CreateRequestID:              info.GetCreateRequestID(),
+		DecisionVersion:              info.GetDecisionVersion(),
+		DecisionScheduleID:           info.GetDecisionScheduleID(),
+		DecisionStartedID:            info.GetDecisionStartedID(),
+		DecisionRequestID:            info.GetDecisionRequestID(),
+		DecisionTimeout:              info.GetDecisionTimeout(),
+		DecisionAttempt:              info.GetDecisionAttempt(),
+		DecisionTimestamp:            info.GetDecisionTimestampNanos(),
```

Comment:
Why StartTimestamp is converted using time.Unix and this one is not?

Reply:
Because the existing persistence interface (which is the input / output to/from the persistence APIs)  passes a time.Time for StartTimestamp and uint64 for DecisionTimestamp